hack/testing/check-curator.sh

```diff
@@ -100,66 +116,108 @@ project-prod:
   delete:
     months: 2
 EOF
-oc delete secret curator-config || echo no such secret curator-config - ignore
-oc secrets new curator-config settings=$curtest
-oc volumes dc/logging-curator --add --type=secret --secret-name=curator-config --mount-path=/etc/curator --name=curator-config --overwrite
-# scale down dc
-oc scale --replicas=0 dc logging-curator
-# wait for pod to go away
-ii=120
-incr=10
-while [ $ii -gt 0 ] ; do
-    if oc describe pod/$curpod > /dev/null ; then
-        echo $curpod still running
-    else
-        break
-    fi
-    sleep $incr
-    ii=`expr $ii - $incr`
-done
-# scale up dc
-oc scale --replicas=1 dc logging-curator
-# wait for pod to start
-ii=120
-incr=10
-while [ $ii -gt 0 ] ; do
-    curpod=`oc get pods | awk -v "pat=^logging-curator${ops}-.* Running" '/-deploy/ {next}; /-build/ {next}; $0 ~ pat {print $1}'`
-    if [ -n "$curpod" ] ; then
-        break
-    fi
-    sleep $incr
-    ii=`expr $ii - $incr`
-done
-# query ES
-curout=`mktemp`
-oc exec $curpod -- curator --host logging-es${ops} --use_ssl --certificate /etc/curator/keys/ca \
-   --client-cert /etc/curator/keys/cert --client-key /etc/curator/keys/key --loglevel ERROR \
-   show indices --all-indices > $curout 2>&1
-# verify that project-dev-yesterday project-qe-lastweek project-prod-fourweeksago .operations-twomonthsago are deleted
-set -- project-dev "$today" project-dev "$yesterday" project-qe "$today" project-qe "$lastweek" project-prod "$today" project-prod "$fourweeksago" .operations "$today" .operations "$twomonthsago" default-index "$today" default-index "$thirtydaysago"
-rc=0
-while [ -n "$1" ] ; do
-    proj="$1" ; shift
-    idx="${proj}.$1"
-    if [ "$1" = "$today" ] ; then
-        # index must be present
-        if grep \^"$idx"\$ $curout > /dev/null 2>&1 ; then
-            echo good - index "$idx" is present
+    oc delete secret curator-config${ops} || echo no such secret curator-config${ops} - ignore
+    oc secrets new curator-config${ops} settings=$curtest
+    oc volumes dc/logging-curator${ops} --add --type=secret --secret-name=curator-config${ops} --mount-path=/etc/curator --name=curator-config --overwrite
+    # scale down dc
+    oc scale --replicas=0 dc logging-curator${ops}
+    # wait for pod to go away
+    ii=120
+    incr=10
+    while [ $ii -gt 0 ] ; do
```

Comment:
you have at least two of these... could they be a function? preferably one that results in an error if you actually time out on the while loop?